src/appleseed.studio/mainwindow/project/entityeditor.cpp

```diff
@@ -484,37 +484,10 @@ unique_ptr<IInputWidgetProxy> EntityEditor::create_colormap_input_widgets(const
 {
     const string name = metadata.get<string>("name");
 
-    double slider_min, slider_max;
-    double validator_min, validator_max;
-
-    if (metadata.dictionaries().exist("min"))
-    {
-        const Dictionary& min = metadata.dictionary("min");
-        slider_min = min.get<double>("value");
-        validator_min = min.get<string>("type") == "hard" ? slider_min : -numeric_limits<double>::max();
-    }
-    else
-    {
-        slider_min = 0.0;
-        validator_min = -numeric_limits<double>::max();
-    }
-
-    if (metadata.dictionaries().exist("max"))
-    {
-        const Dictionary& max = metadata.dictionary("max");
-        slider_max = max.get<double>("value");
-        validator_max = max.get<string>("type") == "hard" ? slider_max : +numeric_limits<double>::max();
-    }
-    else
-    {
-        slider_max = 1.0;
-        validator_max = +numeric_limits<double>::max();
-    }
-
     const QString default_value = metadata.strings().exist("default") ? metadata.get<QString>("default") : "";
 
     ColorMapInputWidget* input_widget = new ColorMapInputWidget(m_parent);
-    input_widget->set_validator(new QDoubleValidatorWithDefault(validator_min, validator_max, 16, default_value));
+    input_widget->set_validator(new QDoubleValidatorWithDefault(-numeric_limits<double>::max(), numeric_limits<double>::max(), 16, default_value));
```

Comment:
You had asked:

> Remove slider_min and slider_max variables and the respective validators

and I said "yes" but I must confess I missed the part about the validators. **The validators must stay.** Only the unused `slider_min` and `slider_max` variables needed to be removed (since they were unused). The logic here has always been completely correct, it's just that a couple variables were unused.